Kernel/Random.cpp

```diff
@@ -63,6 +65,17 @@ KernelRng::KernelRng()
 
             this->resource().add_random_event(value, i % 32);
         }
+    } else {
+        // Add HPET as entropy source if we don't have anything better.
+        klog() << "KernelRng: Using HPET as entropy source (bad!)";
+
+        u64 seconds_since_boot;
+        u32 ticks_this_second;
+        u64 hpet_time = HPET::the().update_time(seconds_since_boot, ticks_this_second, true);
+        for (size_t i = 0; i < resource().pool_count * resource().reseed_threshold; ++i) {
+            this->resource().add_random_event(hpet_time, i % 32);
```

Comment:
Please do not call `HPET::update_time` for this purpose. Instead, add a new method that reads the main counter without a loop, and do this maybe inside this loop. Also, `% 32` essentially just discards the upper 64 bits, so you might as well just read the lower 32 bits. Or you just feed the entire 64 bit value...

Reply:
Done.